r3bbase/R3BFileSource.cxx

```diff
@@ -370,38 +427,49 @@ void R3BFileSource::SetInTree(TTree* tempTree)
     Init();
 }
 
+// ----  Method getrunid -----------------------------------------------------
+Int_t R3BFileSource::getrunid(uint64_t st)
+{
+    for (Int_t i = 0; i < array_size - 1; i++)
+        if (st >= ftimestamp[i] && st < ftimestamp[i + 1])
+        {
+            nextts = ftimestamp[i + 1];
+            LOG(DEBUG) << "New timestamp " << nextts << " for runid " << frunid[i];
+            return frunid[i];
+        }
+
+    if (st >= ftimestamp[array_size - 1])
+    {
+        nextts = ftimestamp[array_size - 1];
+        LOG(DEBUG) << "Final timestamp " << nextts << " for runid " << frunid[array_size - 1];
+        return frunid[array_size - 1];
+    }
+
+    if (nextts > 0)
+        LOG(WARNING) << "R3BFileSource::ReadEvent() RunId was not found, it will be 1";
+    nextts = -1;
+
+    return 1;
+}
+
 // ----  Method ReadEvent ----------------------------------------------------
 Int_t R3BFileSource::ReadEvent(UInt_t i)
 {
     /** TODO
      ** We should use here the timestamp from the header to look for the right runId
      ** and set up the parameters for the analysis of the root files.
      ** std::cout << fEvtHeader->GetTimeStamp() << std::endl;
-     ** Below an example with the event number, just for tests.
      **/
 
-    if (i > fEntryMax && fEntryMax != -1 && fInputFile.is_open())
+    if (nextts == 0 && fEvtHeader->GetTimeStamp() > nextts)
```

Comment:
Can't one contract these two `if` and `if else` blocks into a single one with the condition `nextts >= 0` ?
Also, what should happen, when the condition is not true? Is there an `else` block that may be helpful?